src/ripple/test/BasicNetwork.h

```diff
@@ -197,10 +199,10 @@ class BasicNetwork
             boost::intrusive::make_multiset<msg,
                 boost::intrusive::constant_time_size<false>>::type;
 
-        std::unordered_map<Peer*, by_to_list> by_to_;
-        std::unordered_map<Peer*, by_from_list> by_from_;
-        by_when_set by_when_;
         qalloc alloc_;
+        by_when_set by_when_;
+        std::unordered_map<Peer, by_to_list> by_to_;
+        std::unordered_map<Peer, by_from_list> by_from_;
```

Comment:
Maybe add that to the doc specifying peer requirements. Peer types should be cheap, and point out that a pointer meets the requirements.